pt_miniscreen/app.py

```diff
@@ -88,59 +98,63 @@ def setup_events(self) -> None:
         )
         subscribe(AppEvent.STOP_SCREENSAVER, lambda: self.pop_tile_group())
 
-        def set_is_user_controlled(user_has_control) -> None:
-            if self.user_has_control and not user_has_control:
-                self.user_gave_back_control_event.set()
+    def set_is_user_controlled(self, user_has_control) -> None:
+        if self.user_has_control and not user_has_control:
+            self.user_gave_back_control_event.set()
+
+        logger.info(
+            f"User has {'taken' if user_has_control else 'given back'} control of the miniscreen"
+        )
+
+    def on_button_released(self):
+        if not self.state_manager.buttons_should_be_handled():
+            logger.info("State manager says that buttons should not be handled")
+            return
+
+        self.state_manager.wake()
+
+    def on_up_button_released(self):
+        logger.debug(f"Handling up button for tile group {self.current_tile_group}")
+        self.on_button_released()
```

Comment:
the early return in `on_button_released` won't prevent the button being handled anymore as intended

Reply:
yeah my bad, I'll fix that 👍